### PR TITLE
Port Constraint samplers

### PR DESCRIPTION
**Pull request by anasarrak**
I've been trying to fix the tests, but these tests depends on the [df2_kdl.h](https://github.com/AcutronicRobotics/geometry2/tree/ros2/tf2_kdl),  I was having the following errors https://gist.github.com/anasarrak/9597153fe76ab6bb24d4002a77d09e6a, this might take us a bit to solve.. I prefer to continue porting the rest of the modules.

### CODE REVIEW

**Comment by vmayoral** [comment]
Fixed CI, let's see results @anasarrak

File: moveit_core/constraint_samplers/src/constraint_sampler_tools.cpp [hunk 1]
```diff
@@ -59,13 +61,17 @@ double constraint_samplers::countSamplesPerSecond(const ConstraintSamplerPtr& sa
...
   {
-    ROS_ERROR_NAMED("constraint_samplers", "No sampler specified for counting samples per second");
+    RCLCPP_ERROR(logger, "No sampler specified for counting samples per second");
     return 0.0;
   }
   robot_state::RobotState ks(reference_state);
   unsigned long int valid = 0;
   unsigned long int total = 0;
-  ros::WallTime end = ros::WallTime::now() + ros::WallDuration(1.0);
+  //TODO RCLCPP::WallTimer has not a .now() function, and there is no wallDuration (they use std::chrono::nanoseconds)
```
Comment (anasarrak) [inline moveit_core/constraint_samplers/src/constraint_sampler_tools.cpp:70]: @vmayoral this is something that we have to ask, they sometimes use ros::time::now() and on this case they use ros::Walltime.
ros::Walltime : _Time representation. Always wall-clock time_
In the Api of ros2, there is a [Walltimer](http://docs.ros2.org/crystal/api/rclcpp/classrclcpp_1_1WallTimer.html) but I think that it has nothing to do with ros::walltime. 

I switched it to rclcpp::Clock().now() for now, we will see if we have to change it...

Comment (vmayoral) [reply moveit_core/constraint_samplers/src/constraint_sampler_tools.cpp:70]: @anasarrak, can you please open an issue in ros-planning and ask for the maintainer's input?

Comment (vmayoral) [reply moveit_core/constraint_samplers/src/constraint_sampler_tools.cpp:70]: For tracking purposes, discussing pushed to https://github.com/ros-planning/moveit2/issues/31

**Comment by vmayoral** [comment]
I'm a bit troubled. CI in OS X seems to be fine however in my local installation I get the following error:

```bash
In file included from /Users/victor/ros2_moveit_ws/src/moveit2/moveit_core/constraint_samplers/src/constraint_sampler.cpp:37:
In file included from /Users/victor/ros2_moveit_ws/src/moveit2/moveit_core/constraint_samplers/include/moveit/constraint_samplers/constraint_sampler.h:40:
In file included from /Users/victor/ros2_moveit_ws/src/moveit2/moveit_core/planning_scene/include/moveit/planning_scene/planning_scene.h:40:
In file included from /Users/victor/ros2_moveit_ws/src/moveit2/moveit_core/robot_model/include/moveit/robot_model/robot_model.h:43:
/Users/victor/ros2_moveit_ws/install/include/srdfdom/model.h:44:10: fatal error: 'urdf/model.h' file not found
#include <urdf/model.h>
         ^~~~~~~~~~~~~~
2 warnings generated.
1 error generated.
make[2]: *** [constraint_samplers/CMakeFiles/moveit_constraint_samplers.dir/src/constraint_sampler.cpp.o] Error 1
make[1]: *** [constraint_samplers/CMakeFiles/moveit_constraint_samplers.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
2 warnings generated.
```

Not the first time this happens. @LanderU, any thoughts on this? I'm using OS X 10.14.3.

**Comment by vmayoral** [comment]
@anasarrak please note https://github.com/AcutronicRobotics/moveit2/pull/14/commits/3c40d36c41088740b08a502a8b943c24c48b38bb.

**Pull request merged**